src/main/java/Miku.java

```diff
@@ -0,0 +1,147 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class Miku {
+
+    private static final String[] TODO_REQUIRED_OPTIONS = {"desc"};
+    private static final String[] DEADLINE_REQUIRED_OPTIONS = {"desc","by"};
+    private static final String[] EVENT_REQUIRED_OPTIONS = {"desc","from","to"};
+
+    public static void main(String[] args) {
+        Scanner inputScanner = new Scanner(System.in);
+        TaskList taskList = new TaskList();
+
+        printGreeting();
+
+        String line = inputScanner.nextLine();
+        printDivider();
+
+        while (!line.equals("bye")) {
```

Comment:
However, there is a lot of usage of magic strings and numbers throughout the codebase.